### PR TITLE
[mnist] Add support for S3 in TensorBoard component; Update docs.

### PR DESCRIPTION
**Pull request by eLco**
Adding support for S3 in Tensorboard component with refactoring similar to Train component.

Updated README for Tensorboard and fixed few typos.

Related to #466

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/examples/499)

### CODE REVIEW

File: mnist/README.md [hunk 1]
```diff
@@ -251,7 +251,7 @@ spec:
        ...
        ```
 
-  1. Next we need to set the environment variable `GOOGLE_APPLICATION_CREDENTIALS` so that our code knows
+  2. Next we need to set the environment variable `GOOGLE_APPLICATION_CREDENTIALS` so that our code knows
```
Comment (jlewi) [inline mnist/README.md]: Why did you change the numbering? We should be letting markdown autonumber the items.

File: mnist/README.md [hunk 2]
```diff
@@ -344,14 +353,14 @@ various environment variables configuring access to S3.
      export S3_VERIFY_SSL=1 #set to 0 for defaul minio installs 
      ```
 
-  1. Create a K8s secret containing your AWS credentials
+  2. Create a K8s secret containing your AWS credentials
```
Comment (jlewi) [inline mnist/README.md]: same comment as above; we should be using markdown auto-numbering.

**Comment by jlewi** [comment]
Looks like you need to fix some lint issues and rebase.

File: mnist/README.md (hunk 1 above)
Comment (eLco) [reply mnist/README.md]: I'm sorry, done by mistake. I will get latest updates from master and remove these changes.

**Comment by eLco** [comment]
> Looks like you need to fix some lint issues and rebase.

Seems like there is an issue with lint test - https://github.com/kubeflow/examples/issues/506

**Comment by texasmichelle** [comment]
It looks like the lint issues were resolved in #493. If you rebase, that should fix the failure.

**Comment by jlewi** [comment]
Looks like the mnist E2E test is failing. I'm guessing its not related to this PR since I saw it was failing in the other mnist PR as well.

**Comment by jlewi** [comment]
If you add the pytest macros to indicate the tests are expected failures (per #494) that should hopefully unblock this PR.

**Comment by jlewi** [comment]
Since the training test is failing at head; I'll go ahead and merge this manually. 
I think if we rewrite 
https://github.com/kubeflow/examples/blob/master/mnist/testing/tfjob_test.py
to use pytest we could potentially mark it as an expected failure
